app/controllers/katello/api/registry/registry_proxies_controller.rb

```diff
@@ -85,6 +84,148 @@ def registry_authorize
+            locals: {message: "Organization not found: '#{org_label}'" })
+          return false
+        end
+        @organization = org.first
+      else
+        org_id = props[:organization]
+        unless org_id.present? && org_id == org_id.to_i.to_s # checks if id string is integer and has no leading zeros
+          render_error('custom_error', status: unprocessable_entity,
+            locals: {message: "Invalid format. Organization id must be an integer without leading zeros." })
+          return false
+        end
+        @organization = Organization.find(org_id.to_i)
+      end
+      return true
+    end
+
+    def check_blob_push_product(props)
+      # find the product using the appropriate schema
+      if props[:schema] == "label"
+        prod_label = props[:product]
+        unless prod_label.present? && prod_label.length > 0
+          render_error('custom_error', status: unprocessable_entity,
+            locals: {message: "Invalid format. Product label cannot be blank." })
+          return false
+        end
+        product = @organization.products.where("LOWER(label) = '#{prod_label}'") # convert to lowercase
+        if product.length > 1 # reject ambiguous products (possible due to lowercase conversion)
+          render_error('custom_error', status: unprocessable_entity,
+            locals: {message: "Product label is ambiguous" })
+          return false
+        elsif product.length == 0
+          render_error('custom_error', status: unprocessable_entity,
+            locals: {message: "Product not found: '#{prod_label}'" })
+          return false
+        end
+        @product = product.first
+      else
+        prod_id = props[:product]
+        unless prod_id.present? && prod_id == prod_id.to_i.to_s # checks if id string is integer and has no leading zeros
+          render_error('custom_error', status: unprocessable_entity,
+            locals: {message: "Invalid format. Product id must be an integer without leading zeros." })
+          return false
+        end
+        @product = @organization.products.find(prod_id.to_i)
+      end
+      return true
+    end
+
+    def check_blob_push_container(props)
+      # All incorrect params should have been rejected by this point, assume ok
+      @container_name = props[:name]
+      @container_path_input = "#{props[:organization]}/#{props[:product]}/#{props[:name]}"
+      @container_path_actual = "#{@organization.label}/#{@product.label}/#{@container_name}"
+      return true
+    end
+
+    def create_container_repo_if_needed
+      if @product.root_repositories.where(label: @container_name).empty?
+        root = @product.add_repo(
+          name: @container_name,
+          label: @container_name,
+          download_policy: 'immediate',
+          content_type: Repository::DOCKER_TYPE,
+          unprotected: true,
+          is_container_push: true
+        )
+        sync_task(::Actions::Katello::Repository::CreateRoot, root, @container_path_input) 
```

Comment:
`::Actions::Katello::Repository::Create` can now be modified to skip the `Pulp3::Orchestration::Repository::Create` and `Katello::Repository::MetadataGenerate` steps if the repository is a push one.

Currently, an empty repo is still being created in Pulp with its href saved to the `::Katello::Repository`.

Reply:
This should now be taken care of. Thank you for the help!